src/pages/map/MapPage/MapPage.tsx

```diff
@@ -189,76 +299,73 @@ const MapPage = () => {
-            {mapSearchData &&
-              mapSearchData.map(m => (
-                <MapMarker
-                  key={`marker__${m.location.latitude}-${m.location.longitude}`}
-                  position={{ lat: m.location.latitude, lng: m.location.longitude }}
-                  onClick={() => {
-                    setIsOverlayOpen(true);
-                    setSelectedPoint({
-                      lat: m.location.latitude,
-                      lng: m.location.longitude,
-                      type: m.category,
-                    });
-                  }}
-                ></MapMarker>
-              ))}
-
-            {/* API 데이터가 없는 경우 panelMockData로 표시 (테스트용, 추후 제거) */}
-            {(!mapSearchData || mapSearchData.length === 0) &&
-              panelMockData.map(m => (
-                <MapMarker
-                  key={`marker__mock__${m.location.latitude}-${m.location.longitude}`}
-                  position={{ lat: m.location.latitude, lng: m.location.longitude }}
-                  onClick={() => {
-                    setIsOverlayOpen(true);
-                    setSelectedPoint({
-                      lat: m.location.latitude,
-                      lng: m.location.longitude,
-                      type: m.category,
-                    });
-                  }}
-                ></MapMarker>
-              ))}
+            {/* 마커 */}
+            {mapItemData &&
+              mapItemData.map(m => {
+                let id: number;
+                let uniqueKey: string;
+
+                if (m.category === MapItemCategory.PROJECT && 'projectId' in m) {
+                  id = m.projectId;
+                  uniqueKey = `marker__project-${m.projectId}`;
+                } else if (m.category === MapItemCategory.STUDY && 'studyId' in m) {
+                  id = m.studyId;
+                  uniqueKey = `marker__study-${m.studyId}`;
+                } else if (m.category === MapItemCategory.PLAYER && 'userId' in m) {
+                  id = m.userId;
+                  uniqueKey = `marker__player-${m.userId}`;
+                } else if (m.category === MapItemCategory.CAFE && 'cafeId' in m) {
+                  id = m.cafeId;
+                  uniqueKey = `marker__cafe-${m.cafeId}`;
+                } else {
+                  return null;
+                }
+
+                return (
+                  <MapMarker
+                    key={uniqueKey}
+                    position={{ lat: m.location.latitude, lng: m.location.longitude }}
+                    onClick={() => {
+                      setIsOverlayOpen(true);
+                      setSelectedPoint({
+                        lat: m.location.latitude,
+                        lng: m.location.longitude,
+                        category: m.category,
+                        id: id,
+                      });
+
+                      // cafe 마커 클릭 시 MapSubPanel도 열기
+                      if (m.category === MapItemCategory.CAFE && 'cafeId' in m) {
```

Comment:
```suggestion
                      if (isMapPanelOpen && m.category === MapItemCategory.CAFE && 'cafeId' in m) {
```

이 부분 때문에 서브패널 오류가 생겼던 것 같습니당